src/vm/datatypes/files.c

```diff
@@ -82,7 +88,10 @@ static Value readFullFile(DictuVM *vm, int argCount, Value *args) {
         buffer = SHRINK_ARRAY(vm, buffer, char, fileSize + 1, bytesRead + 1);
     }
 
-    buffer[bytesRead] = '\0';
+    if (strcmp(file->openType, "rb") != 0) {
+        buffer[bytesRead] = '\0';
+    }
```

Comment:
We will want to null terminate here as Dictu expects all strings to be null terminated. Is there a reason you didn't, did you find something break or?